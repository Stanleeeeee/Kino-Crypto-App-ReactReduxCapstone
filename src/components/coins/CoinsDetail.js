import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import './coins.css';

const CoinsDetail = () => {
  const { coins } = useSelector((state) => state.coins);
  const { active } = useSelector((state) => state.coins);
  return (
    <div className="detailContainer">
      <NavLink className="backBttn" to="/">
        <img alt="Back button" src="https://img.icons8.com/ios-glyphs/30/ffffff/circled-left-2.png" />
      </NavLink>
      <p>Cryptocurrency</p>
      <div className="infoContainer">
        <p className="detailInfo">
          {`${coins[active].coin_id} `}
          -
          {` ${coins[active].coin_name}`}
        </p>
      </div>
      <p>Value</p>
      <div className="infoContainer">
        <p className="detailInfo">{`$${coins[active].coin_value}`}</p>
      </div>
      <p>Daily Fluctuation</p>
      <div className="infoContainer">
        <p className="detailInfo">{`${coins[active].coin_change}%`}</p>
      </div>
    </div>
  );
};
export default CoinsDetail;
