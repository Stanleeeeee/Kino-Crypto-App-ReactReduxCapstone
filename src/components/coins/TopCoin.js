import { useSelector } from 'react-redux';
import './topCoin.css';

const TopCoin = () => {
  const { coins } = useSelector((state) => state.coins);

  return (
    <>
      <div className="container">
        <h3>
          Most Valuable -
          {' '}
          { coins !== undefined ? coins[0].coin_name : <p>Loading Crypto</p> }
        </h3>
        <div className="content">
          <img alt="BTC" src="https://img.icons8.com/color/70/000000/bitcoin--v3.png" />
          { coins !== undefined ? `$${coins[0].coin_value}` : <p>Loading Value</p> }
        </div>
      </div>
    </>
  );
};

export default TopCoin;
