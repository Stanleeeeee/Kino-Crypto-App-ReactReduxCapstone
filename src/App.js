import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Navigation from './components/navigation/Navigation';
import Coins from './components/coins/Coins';
import TopCoin from './components/coins/TopCoin';
import CoinsDetail from './components/coins/CoinsDetail';

function App() {
  return (
    <div>
      <>
        <Router>
          <Navigation />
          <TopCoin />
          <Routes>
            <Route path="/" element={<Coins />} />
            <Route path="/TopCoin" element={<TopCoin />} />
            <Route path="/coin-detail" element={<CoinsDetail />} />
          </Routes>
        </Router>
      </>
    </div>
  );
}

export default App;
