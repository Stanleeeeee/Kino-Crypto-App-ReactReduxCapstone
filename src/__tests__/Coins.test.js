import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import Coins from '../components/coins/Coins';
import App from '../App';

import store from '../redux/configureStore';
import { fetchGetCoins } from '../redux/coins';

describe('pages test snapshots', () => {
  it('testing detail page', async () => {
    await store.dispatch(fetchGetCoins());
    const coinsPage = renderer
      .create(
        <Provider store={store}>
          <App>
            <Coins />
          </App>
        </Provider>,
      )
      .toJSON();
    expect(coinsPage).toMatchSnapshot();
  });
});
