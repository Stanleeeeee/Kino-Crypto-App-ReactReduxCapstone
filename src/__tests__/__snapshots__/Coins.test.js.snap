// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages test snapshots testing detail page 1`] = `
<div>
  <header>
    <img
      alt="Mobile menu button"
      src="https://img.icons8.com/material-rounded/24/ffffff/menu--v1.png"
    />
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Kino Crypto 
      </h1>
    </a>
    <div>
      <img
        alt="config button"
        src="https://img.icons8.com/external-those-icons-fill-those-icons/24/ffffff/external-gear-settings-those-icons-fill-those-icons-1.png"
      />
    </div>
  </header>
  <div
    className="container"
  >
    <h3>
      Most Valuable -
       
      BTC
    </h3>
    <div
      className="content"
    >
      <img
        alt="BTC"
        src="https://img.icons8.com/color/70/000000/bitcoin--v3.png"
      />
      $21840.28
    </div>
  </div>
  <div
    className="coinsContainer"
  >
    <div
      className="filtersContainer"
    >
      <p>
        Top 10 Most Valuable Crypto
      </p>
      <select
        onChange={[Function]}
        value="value"
      >
        <option
          value="value"
        >
          value
        </option>
        <option
          value="fluctuation"
        >
          fluctuation
        </option>
      </select>
    </div>
    <ul
      className="coinBox"
    >
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Bitcoin 
              -
               BTC
            </p>
            <p>
              $21840.28
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Ethereum 
              -
               ETH
            </p>
            <p>
              $1226.371
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Tether 
              -
               USDT
            </p>
            <p>
              $0.999495
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              USD Coin 
              -
               USDC
            </p>
            <p>
              $0.999448
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              BNB 
              -
               BNB
            </p>
            <p>
              $243.8945
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Binance USD 
              -
               BUSD
            </p>
            <p>
              $1.000417
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Cardano 
              -
               ADA
            </p>
            <p>
              $0.481701
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              XRP 
              -
               XRP
            </p>
            <p>
              $0.345850
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Solana 
              -
               SOL
            </p>
            <p>
              $38.53866
            </p>
          </div>
        </li>
      </a>
      <a
        className="route"
        href="/coin-detail"
        onClick={[Function]}
      >
        <li>
          <div
            className="coinItem"
          >
            <img
              alt="currency"
              src="https://img.icons8.com/fluency/35/000000/stack-of-coins.png"
            />
            <p>
              Dogecoin 
              -
               DOGE
            </p>
            <p>
              $0.069892
            </p>
          </div>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
